sklearn/ensemble/partial_dependence.py

```diff
@@ -306,8 +306,8 @@ def convert_feature(fx):
                 l.append(feature_names[i])
             names.append(l)
     except IndexError:
-        raise ValueError('features[i] must be in [0, n_features) '
-                         'but was %d' % i)
+        raise ValueError('features[i] must be in [0, len(feature_names)={0} ) '
+                         'but was {1}'.format(len(feature_names), i))
```

Comment:
thanks for linking that to me. yeah, I agree that the error message is mostly there. It just seemed extremely ambiguous to me when I first read it (which made me question its value to a user), but it's obvious in hindsight after I've seen the code in question.